docker/Dockerfile

```diff
@@ -1,107 +1,122 @@
-# syntax=docker/dockerfile:1
-
-ARG PYTHON_VERSION=3.9
-##################
-##  base image  ##
-##################
-FROM --platform=${TARGETPLATFORM} python:${PYTHON_VERSION}-slim AS python-base
-
-LABEL org.opencontainers.image.authors="mauwii@outlook.de"
-
-# Prepare apt for buildkit cache
-RUN rm -f /etc/apt/apt.conf.d/docker-clean \
-  && echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' >/etc/apt/apt.conf.d/keep-cache
-
-# Install dependencies
-RUN \
-  --mount=type=cache,target=/var/cache/apt,sharing=locked \
-  --mount=type=cache,target=/var/lib/apt,sharing=locked \
-  apt-get update \
-  && apt-get install -y \
-    --no-install-recommends \
-    libgl1-mesa-glx=20.3.* \
-    libglib2.0-0=2.66.* \
-    libopencv-dev=4.5.*
-
-# Set working directory and env
-ARG APPDIR=/usr/src
-ARG APPNAME=InvokeAI
-WORKDIR ${APPDIR}
-ENV PATH ${APPDIR}/${APPNAME}/bin:$PATH
-# Keeps Python from generating .pyc files in the container
-ENV PYTHONDONTWRITEBYTECODE 1
-# Turns off buffering for easier container logging
-ENV PYTHONUNBUFFERED 1
-# Don't fall back to legacy build system
-ENV PIP_USE_PEP517=1
-
-#######################
-##  build pyproject  ##
-#######################
-FROM python-base AS pyproject-builder
-
-# Install build dependencies
-RUN \
-  --mount=type=cache,target=/var/cache/apt,sharing=locked \
-  --mount=type=cache,target=/var/lib/apt,sharing=locked \
-  apt-get update \
-  && apt-get install -y \
-    --no-install-recommends \
-    build-essential=12.9 \
-    gcc=4:10.2.* \
-    python3-dev=3.9.*
-
-# Prepare pip for buildkit cache
-ARG PIP_CACHE_DIR=/var/cache/buildkit/pip
-ENV PIP_CACHE_DIR ${PIP_CACHE_DIR}
-RUN mkdir -p ${PIP_CACHE_DIR}
-
-# Create virtual environment
-RUN --mount=type=cache,target=${PIP_CACHE_DIR} \
-  python3 -m venv "${APPNAME}" \
-  --upgrade-deps
-
-# Install requirements
-COPY --link pyproject.toml .
-COPY --link invokeai/version/invokeai_version.py invokeai/version/__init__.py invokeai/version/
-ARG PIP_EXTRA_INDEX_URL
-ENV PIP_EXTRA_INDEX_URL ${PIP_EXTRA_INDEX_URL}
-RUN --mount=type=cache,target=${PIP_CACHE_DIR} \
-  "${APPNAME}"/bin/pip install .
-
-# Install pyproject.toml
-COPY --link . .
-RUN --mount=type=cache,target=${PIP_CACHE_DIR} \
-  "${APPNAME}/bin/pip" install .
-
-# Build patchmatch
+# syntax=docker/dockerfile:1.4
+
+## Builder stage
+
+FROM ubuntu:22.04 AS builder
+
+ARG DEBIAN_FRONTEND=noninteractive
+RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    apt update && apt-get install -y \
+        git \
+        python3-venv \
```

Comment:
How are we ensuring it doesn't install python 3.11+?

Reply:
`python3-venv` will match the installed python version. we can pin the version to be sure but i don't think it's necessary